server/controllers/finance/reports/financial.patient.js

```diff
@@ -41,13 +43,29 @@ function build(req, res, next) {
   }
 
   const data = {};
+  data.includeStockDistributed = parseInt(options.include_stock_distributed, 10);
 
   return Patients.lookupPatient(req.params.uuid)
     .then(patient => {
       _.extend(data, { patient });
-      return Debtors.getFinancialActivity(patient.debtor_uuid, true);
+      const dbPromises = [
+        Debtors.getFinancialActivity(patient.debtor_uuid, true),
+      ];
+
+      if (data.includeStockDistributed) {
+        dbPromises.push(Patients.stockMovementByPatient(req.params.uuid));
+      }
+
+      return q.all(dbPromises);
```

Comment:
On the server, you don't really need to use `q` anymore.  You can use `Promise.all()`.  BHIMA doesn't support any old node versions without `Promise` support.